common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/model/codegen/BeanGenerator.java

```diff
@@ -0,0 +1,284 @@
+            methdBuilder.addAnnotation(JsonAnyGetter.class);
+        } else {
+            final String attributePrefix = isBoolean(fieldSpec) ? "is" : "get";
+            methdBuilder = MethodSpec.methodBuilder(attributePrefix + getPropertyMethodNameSuffix(property))
+                    .addAnnotation(AnnotationSpec.builder(JsonProperty.class)
+                            .addMember("value", "$S", property.getName()).build());
+        }
+        methdBuilder.addModifiers(Modifier.PUBLIC)
+                .returns(fieldSpec.type)
+                .addCode("return " + fieldSpec.name + ";\n");
+        return methdBuilder.build();
+    }
+
+
+    private static String getPropertyMethodNameSuffix(final Property property) {
+        final String javaPropertyName = Arrays.stream(property.getName().split("_"))
+                .map(StringUtils::capitalize)
+                .collect(Collectors.joining());
+
+        return javaPropertyName;
+    }
+
+    private static boolean isBoolean(FieldSpec fieldSpec) {
+        return fieldSpec.type.equals(ClassName.BOOLEAN.box()) || fieldSpec.type.equals(ClassName.BOOLEAN);
+    }
+
+    public Path getPath(JavaFileObject javaFile, Path outputFolder) {
+        return Paths.get(outputFolder.toString(), javaFile.getName());
+    }
+
+    protected Stream<AnnotationSpec> getDiscriminatorAnnotations(final ObjectType object) {
+        if (StringUtils.isEmpty(object.getDiscriminator()) || object.getSubTypes().isEmpty()) {
+            return Stream.empty();
+        }
+        final AnnotationSpec jsonTypeInfoAnnotation = AnnotationSpec.builder(JsonTypeInfo.class)
+                .addMember("use", "$T.NAME", JsonTypeInfo.Id.class)
+                .addMember("include", "$T.PROPERTY", JsonTypeInfo.As.class)
+                .addMember("property", "$S", object.getDiscriminator())
+                .addMember("visible", "true")
+                .build();
+
+        CodeBlock.Builder annotationBodyBuilder = CodeBlock.builder();
+        List<ObjectType> children = getSubtypes(object).collect(Collectors.toList());
+        if (!children.isEmpty()) {
+            for (int i = 0; i < children.size(); i++) {
+                annotationBodyBuilder.add(
+                        (i == 0 ? "{" : ",") +
+                                "\n@$T(value = $T.class, name = $S)"
+                                + (i == children.size() - 1 ? "\n}" : "")
+                        , JsonSubTypes.Type.class, getClassName(getPackagePrefix(), children.get(i)), children.get(i).getDiscriminatorValue());
+            }
+        }
+        final AnnotationSpec jsonSubTypesAnnotation = AnnotationSpec.builder(JsonSubTypes.class)
+                .addMember("value", annotationBodyBuilder.build())
+                .build();
+
+        return Stream.of(jsonSubTypesAnnotation, jsonTypeInfoAnnotation);
+    }
+
+    protected Stream<AnnotationSpec> getAdditionalTypeAnnotations(final ObjectType object) {
```

Comment:
Here we should rename the parameter to `objectType` too.

Reply:
nice catch 👍